httpx/_models.py

```diff
@@ -1008,7 +1008,7 @@ async def aclose(self) -> None:
                 await self.stream.aclose()
 
 
-class Cookies(typing.MutableMapping[str, str]):
+class Cookies(typing.MutableMapping[str, typing.Optional[str]]):
```

Comment:
```suggestion
class Cookies(typing.MutableMapping[str, str | None]):
```

Reply:
Hi @T-256 ! Actually, That was my initial attempt, but the test workflow [failed](https://github.com/encode/httpx/actions/runs/7914423735/job/21604123090) with this notation.

Reply:
Interesting, no problem then...